llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -603,16 +889,82 @@ Expected<uint64_t> DebugInfoSectionRef::materialize(MCCASReader &Reader,
   SmallVector<char, 512> SectionContents;
   raw_svector_ostream SectionStream(SectionContents);
   unsigned Size = 0;
+  SmallVector<SmallVector<char>> CUContents;
   for (auto [AbbrevOffset, CUData] : LoadedSection->getOffsetAndCUDataRange()) {
     // Copy the data so that we can modify the abbrev offset prior to printing.
     // FIXME: do this with a zero-copy strategy.
     auto MutableCUData = to_vector(CUData);
     if (auto E = getAndSetDebugAbbrevOffsetAndSkip(
-            MutableCUData, Reader.getEndian(), AbbrevOffset);
+            MutableCUData, Reader.getEndian(), AbbrevOffset, false);
         !E)
       return E.takeError();
-    SectionStream << MutableCUData;
-    Size += CUData.size();
+    CUContents.push_back(MutableCUData);
+  }
+
+  InMemoryCASDWARFObject CASObj(AbbrevSectionContents,
+                                Reader.getEndian() == support::little);
+  auto DWARFObj = std::make_unique<InMemoryCASDWARFObject>(CASObj);
+  auto DWARFContextHolder = std::make_unique<DWARFContext>(std::move(DWARFObj));
+  auto *DWARFCtx = DWARFContextHolder.get();
+  DWARFDebugAbbrev Abbrev;
+  Abbrev.extract(DataExtractor(toStringRef(AbbrevSectionContents),
+                               Reader.getEndian() == support::little, 8));
+  uint64_t FixupOffset = 0;
+  for (unsigned I = 0; I < CUContents.size(); I++) {
```

Comment:
We don't use `I` for anything here, but this variable gets shadowed on line 927!

Let's use a range-for loop instead:

`for (auto CUData : CUContents)`

Reply:
ARGH! I had it as a range based for loop but then I changed it because I was trying to see which Compile Unit I was looking at when I got the 8500 stack frames and forgot to change it back!